DMC Config Updates

scottcanoe opened the pull request:
A few updates to DMC configs.

**Content**
 - Using `max_nneighbors=10`
 - Using a new logger handler used for storing evidence, also heavily used for visualization configs. Also discarding the custom class previously used for selective evidence logging.
 - Added several configs to `visualizations.py`
 - Updates to docstrings and README.

**Style/Cosmetic**
 - Discarding unneeded dataloader args.
 - Some small changes to make patterns more consistent across files.
 - Also some config-changing functions that were only used once have been removed, and the necessary changes made directly. This is partly an effort to start winding down usage of functions that make the code unnecessarily complex. I expect there may be a couple more things like this that can get simplified down the road.
 
I expect there may be a few tweaks or additions to visualizations.py later on, but this is pretty close to a final version of the configs.

nielsleadholm reviewed (commented):
Awesome, thanks for making these changes Scott. Just added a few comments.

scottcanoe commented on dmc/configs/README.md:
```diff
@@ -27,7 +28,25 @@ Unless specified otherwise, the following figures/experiments use:
+
+**Notes:**
+- Although evaluating on 10 objects, the model is trained on 77 objects.
+- We need to run this experiment with SELECTIVE logging on so we get the evidence values to analyze.
```
Good point, added.

scottcanoe commented on dmc/configs/common.py:
```diff
@@ -375,129 +380,348 @@ def add_sensor_noise(
+    template: Mapping,
+    run_name: Optional[str] = None,
+) -> Mapping:
+    """Create a copy of an experiment config that uses the 5 "random" rotations.
```
OK, I've changed to 'predefined random rotations' (here and a couple other spots) and also added some explanation to the README. Do you think that's sufficient?

scottcanoe commented on dmc/configs/fig3_robust_sensorimotor_inference.py:
```diff
@@ -95,21 +96,25 @@
+)
+
+# - Random rotation variant (14 random rotations) and sensor noise
+dist_agent_1lm_randrot_all_noise = make_noise_variant(
```
This one only needs to add noise since the copied config uses the 14 random rotations (using `RandomRotationObjectInitializer`).

`make_randrot_noise_variant` could be extended to do add a random rotation object initializer and 14 epochs, but this is the only config that would use it, so I just set it directly here.

nielsleadholm replied:
Makes sense, thanks

scottcanoe commented on dmc/configs/fig9_structured_object_representations.py:
```diff
@@ -23,193 +23,17 @@
  - Probably best run in serial.
 """
 
```
It's been renamed to `fig4_structured_object_representations.py`, but we still need the config unless I'm missing something.

scottcanoe replied:
Or maybe this experiment would be better off as a short experiment in visualizations.py?

nielsleadholm replied:
Ah I missed the fine renaming, makes sense.

scottcanoe commented on dmc/configs/pretraining_experiments.py:
```diff
@@ -444,7 +455,7 @@ def make_10distinctobj_variant(template: dict) -> dict:
 )
 
-# Make 10distinctobj variants
+# Make 10distinctobj variants. Used in `fig8.py` eval experiments.
```
Yes and no - only the multimodal transfer experiments use the 10distinctobj models, but the file name is/was wrong. I've moved multimodal transfer experiments to `supplemental_fig9_multimodal_transfer.py` and modified this comment to reflect it.

scottcanoe commented on dmc/configs/visualizations.py:
```diff
@@ -12,38 +12,187 @@
+- `fig3_evidence_run`: A one-episode experiment used to collect evidence
+   and sensor data for every step. The output is read and plotted by functions in
+    `scripts/fig3.py`.
+- `fig3_symmetry_run`: Runs `dist_agent_1lm_randrot_noise` with storage of
```
Yep, good point. Done.

scottcanoe commented:
Thanks @hlee9212 and @nielsleadholm! 

So it seemed like changing references in the README but not actually changing file names (which we had planned to do later) was a bit sloppy, so I've gone ahead and changed file names to reflect the new figure naming/ordering. That'll probably make most comments "outdated", but I have incorporated all suggested comments. Sorry about that, should have done the renaming out the gate. It seemed cleaner to rename files in a separate commit, but in reality, having README and other comments out-of-sync with the files wasn't great. I believe new names and references are all accurate.

nielsleadholm reviewed (approved):
Awesome, thanks for those changes. Looks great.

vkakerbeck reviewed (approved):
Nice! Great cleanups and additions. It's fun to see this come together. No blocking issues from my side.

scottcanoe commented on dmc/configs/README.md:
```diff
@@ -6,28 +6,49 @@ Below is a summary of configs that correspond to figures in the Demonstrating Mo
 Unless specified otherwise, the following figures/experiments use:
 - 77 objects
-- 5 random rotations
+- 5 predefined "random" rotations. These rotations were randomly generated but are kept constant across experiments. 
```
That's a good question. The results are generally pretty consistent with the 14 random rotations versions, but it's good point. At some point it'd be worth rerunning with a different set of 5.

scottcanoe commented:
OK, thanks for everyone's eagle eyes.
